docs/devsetup/java/maven.md

```diff
@@ -0,0 +1,229 @@
+---
+layout: default
+title: Maven Setup for Java
+grand_parent: Setup for Contributors
+parent: Java Setup
+---
+
+<!--
+© 2016 and later: Unicode, Inc. and others.
+License & terms of use: http://www.unicode.org/copyright.html
+-->
+
+# Maven Setup for Java
+{: .no_toc }
+
+## Contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+---
+
+
+Maven is a pretty standard build tool in the Java ecosystem with a very well-defined
+preferred way to organize the directory structure and execute builds.
+Thus, many IDEs for Java will have some level of support for Maven.
+The command line invocation is still, of course, the standard of truth for the build.
+
+## Installation
+
+Install Maven from your OS package manager.
+The minimum version is 3.2.5, as configured in the `<ICU>/icu4j/pom.xml` root POM file's maven-enforcer-plugin.
+The Maven page lists [system requirements](https://maven.apache.org/download.cgi),
+which includes a minimum of JDK 8.
+
+## IDE Setup
+
+Users of IDEs should familiarize themselves with the information about how to use Maven at
+the command line,
+which also includes information about Maven builds.
+
+> :point_right: **Note**: In most IDEs, whenever the Maven configurations change (`pom.xml` files),
+you will need to refresh your IDE project using the IDE's Maven plugin/functionality.
+
+Most IDEs represent user settings for different codebases via an IDE-specific notion
+usually called "project" or "workspace".
+The IDE's project/workspace is a separate construct from the Maven configurations for the codebase.
+Therefore, creating a new project for an existing codebase is done by "importing" the codebase,
+in which the IDE creates the project/workspace settings files.
+If you pull updates to the upstream ICU codebase codebase that result in changes to Maven settings,
+then the IDE behavior may be behind until you update your IDE project accordingly.
+
+
+### IntelliJ
+
+IntelliJ does a good job of understanding multi-module Maven projects, 
+including the non-standard configuration here.
+It also recognizes the customized locations of source code files and test code files in the configuration here.
+
+To import into IntelliJ:
+
+1. In IntelliJ, open a new project. 
+  a. Recent versions of IntelliJ provide a dialog box on startup to select a project. Click the "Open" button.)
+2. Select the root `pom.xml` in ICU4J (ex: `<ICU>/icu4j/pom.xml`)
```

Comment:
I tried it, and IntelliJ gets really confused (and fails to build) because of `icu4j/tools/utilities-for-cldr/`. Which is that ugly hack to build a shared lib for cldr by picking and choosing files from other places.

Until we clean that I have a workaround. There is a way to disable / ignore a project (in this case `tools/utilities-for-cldr`)
But we need to document that.

---

Might also be an idea to have separate documents for various IDEs (Eclipse, IntelliJ, maybe VSCode?)

Because I anticipate these are areas that will grow.

For example:
- recommended plugins and how to configure them (for example findbugs, checkstyle, etc)
- import settings (compiler, formatting, organizing imports, plugins, etc)
- import maven project
- refresh IDE project from maven projects

Reply:
1. Done.
2. Let's refactor when we receive contributions on the docs to add this info. Agree that it is important & needed.